src/main/java/com/google/devtools/build/lib/packages/SkylarkNativeModule.java

```diff
@@ -118,7 +119,18 @@
       throw new EvalException(loc, "illegal argument in call to glob", e);
     }
 
-    return StarlarkList.copyOf(thread.mutability(), matches);
+    String[] result = new String[matches.size()];
+    for (int i = 0; i < result.length; i++) {
+      String match = matches.get(i);
+      if (match.charAt(0) == '@') {
+        // Add explicit colon to disambiguate from external repository.
+        match = ":" + match;
+      }
+      result[i] = match;
+    }
+    Arrays.sort(result);
```

Comment:
Minor performance concern: If the glob is a Skyframe-cache-hit, then matches will be already sorted. I know that TimSort is good on already sorted arrays, but still... 

Sorry for lack of concrete suggestion; I'm pretty busy today. I'll try to take another look next week (will involve reading over all the code).

Reply:
`PackageFunction.HybridToken.resolveAsync`, which isn't cached by Skyframe, does the sorting of Skyframe results. So, unless I'm misunderstanding, this doesn't actually change the number of sorts of cached globs.

Reply:
> [me in previous github comment] If the glob is a Skyframe-cache-hit, then matches will be already sorted

Er, this may wrong, sorry. PackageFunction.HybridToken.resolve had (past-tense b/c you're deleting it in this commit) a comment "Skyframe glob results are unsorted". I wrote this years ago and may not be true now (or ever). Someone should think about this (I can next week if I have time).

> [benjaminp] So, unless I'm misunderstanding, this doesn't actually change the number of sorts of cached globs.

Yes you're correct. Thanks for the correction.